x-pack/plugins/timelines/common/api/search_strategy/model/language.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from 'zod';
+
+export const language = z.union([z.literal('eql'), z.literal('kuery'), z.literal('lucene')]);
```

Comment:
Awesome! Going to have to add `esql` to this soon as well. Same comment about kibana platform here